File: ruby/lib/phonenumber_ae.rb
# frozen_string_literal: true

# phonenumber_ae — validate, parse and format international phone numbers.
#
# A thin Fiddle binding over one shared native engine (pure Aether, compiled
# from Google libphonenumber's own metadata), the same artifact every other
# language binding in this monorepo uses. Cross-language behaviour is therefore
# identical by construction, not by test. No phone-number logic lives in this
# gem; it only marshals values across the C ABI (v6, full PhoneNumberUtil parity
# plus ShortNumberInfo, TimeZones, Carrier and Geocoder).
#
#     require "phonenumber_ae"
#
#     num = PhoneNumberAe.parse("+1 650 253 0000", "US")
#     num.national_number                       # => "6502530000"
#     PhoneNumberAe.is_valid_number("US", "+1 650 253 0000")   # => true
#     PhoneNumberAe.format("US", "6502530000", PhoneNumberAe::INTERNATIONAL)
#     # => "+1 650-253-0000"
#
#     ayt = PhoneNumberAe::AsYouTypeFormatter.new("US")
#     "6502530000".each_char { |c| ayt.input_digit(c) }
#     ayt.result                                # => "(650) 253-0000"
#
#     PhoneNumberAe.find_numbers("call 201-555-0123 today", "US")

require_relative "phonenumber_ae/version"
require_relative "phonenumber_ae/native"
require_relative "phonenumber_ae/phone_number"

File: dart/lib/phonenumber_ae.dart
/// Validate, parse and format international phone numbers (ABI v6).
///
/// A thin `dart:ffi` binding over the monorepo's one shared native engine
/// (`core/native/libphonenumber_ae.so`, compiled from pure Aether over Google
/// libphonenumber's own metadata). It contains **no phone-number logic**: every
/// member marshals to an `aether_pn_embed_*` call. One engine, one set of
/// behaviours, N language surfaces.
///
/// ```dart
/// import 'package:phonenumber_ae/phonenumber_ae.dart' as pn;
///
/// final num = pn.parse('+1 650 253 0000', 'US');
/// num.nationalNumber;                                     // '6502530000'
/// pn.isValidNumber('US', '+1 201 555 0123');              // true
/// pn.format('US', '2015550123', pn.PhoneFormat.international); // '+1 201-555-0123'
///
/// final ayt = pn.AsYouTypeFormatter('US');
/// for (final c in '6502530000'.split('')) ayt.inputDigit(c);   // '(650) 253-0000'
///
/// pn.findNumbers('call 201-555-0123 today', 'US');
/// ```
library;

export 'src/phonenumber.dart'
    show
        // enums / constant groups
        PhoneFormat,
        PhoneNumberType,
        ValidationResult,
        MatchType,
        CountryCodeSource,
        Leniency,
        ShortNumberCost,
        // metadata
        countryCode,
        exampleNumber,
        exampleNumberForType,
        invalidExampleNumber,
        possibleLengths,
        regionCodeForCountryCode,
        isNanpaCountry,
        nddPrefixForRegion,
        regions,
        ccRegionCount,
        regionsForCountryCode,
        // parse
        ParsedNumber,
        parse,
        nationalNumber,
        // validation
        isPossibleNumber,
        isPossibleNumberWithReason,
        isValidNumber,
        isValidNumberForRegion,
        numberType,
        canBeInternationallyDialled,
        // formatting
        format,
        formatNational,
        formatInternational,
        formatE164,
        formatRfc3966,
        formatOutOfCountry,
        // helpers
        isNumberMatch,
        truncateTooLong,
        normalizeDigitsOnly,
        convertAlphaCharacters,
        isAlphaNumber,
        abiVersion,
        nativeLibraryPath,
        // stateful / matcher
        AsYouTypeFormatter,
        PhoneNumberMatch,
        findNumbers,
        // short / emergency numbers
        ShortNumberInfo,
        // timezone + carrier + geocoder lookup
        TimeZones,
        Carrier,
        Geocoder;
